static/js/settings_account.js

```diff
@@ -459,12 +459,29 @@ export function set_up() {
         e.preventDefault();
         e.stopPropagation();
         const change_password_error = $("#change_password_modal").find("#dialog_error");
+        change_password_error.hide();
 
         const data = {
             old_password: $("#old_password").val(),
             new_password: $("#new_password").val(),
         };
 
+        function show_error_message(rendered_error_msg) {
```

Comment:
Could this function be replaced with `ui_report.error`?  I guess the problem is we've already started the dialog spinner -- arguably the issue here is that we should be calling a validation function to check these values before initializing the spinner, which would require extending the `dialog_widget` class to support doing so?

Reply:
Yes we can extend the `dialog_widget` class to have something like `validate_input` which will be first do the checks that we can do at client level itself and then the rest of the code for spinner and making request can be executed.